qp/pdf.py

```diff
@@ -55,18 +60,54 @@ def quantize(self, percent=10.0, number=None):
         print("Result: ", self.quantiles)
         return self.quantiles
 
-    def interpolate(self):
+    def interpolate(self, number=100):
+        """
+        Interpolates the quantiles.
 
-        return
+        Parameters
+        ----------
+        number: int
+            The number of points over which to interpolate, bounded by the quantile value endpoints
+
+        Returns
+        -------
+        grid: ndarray, float
+            The input grid upon which to interpolate
+        interpolated : ndarray, float
+            The interpolated points.
+
+        Notes
+        -----
 
-    def plot(self, limits):
+        """
+        if self.interpolator is None:
+
+            if self.quantiles is None:
+                self.quantiles = self.quantize()
+            self.difs = self.quantiles[1:]-self.quantiles[:-1]
+            self.mids = (self.quantiles[1:]+self.quantiles[:-1])/2.
```

Comment:
Hmm. What about the PDF beyond the lowest and highest quantile points? Would it make sense to define `minusinfinity` and `plusinfinity` and "interpolate" (actually, extrapolate) out to them? I'm not sure - but the documentation needs to give good advice, I think. 

Related to this question is: this method returns a _tabulated function_, but usually what you want is a _function_. This would take as input an `x` value, and return the interpolated value of the quantile-approximated density at that `x` point. Maybe this is a separate method, called `approximate(x)`? It looks like that `approximate` method could just return the value of `self.interpolator(x)` - is that correct? Seems like a good addition to me.